benchmarks/io.py

```diff
@@ -1,59 +1,109 @@
+        lambda filename: read_file(filename, driver="GeoJSON"),
+    ),
+    "Parquet": (
+        ".parquet",
+        lambda gdf, filename: gdf.to_parquet(filename),
+        lambda filename: read_parquet(filename),
+    ),
+    "Feather": (
+        ".feather",
+        lambda gdf, filename: gdf.to_feather(filename),
+        lambda filename: read_feather(filename),
+    ),
+}
 
-    def setup(self, ext):
 
-        self.driver_dict = {".shp": "ESRI Shapefile",
-                            ".json": "GeoJSON",
-                            ".gpkg": "GPKG"}
-        driver = self.driver_dict[ext]
+class Bench:
+
+    params = ["ESRI Shapefile", "GeoJSON", "GPKG", "Parquet", "Feather"]
+    param_names = ["file_format"]
+
+    def setup(self, file_format):
+        self.ext, self.writer, self.reader = format_dict[file_format]
 
         num_points = 20000
         xs = np.random.rand(num_points)
         ys = np.random.rand(num_points)
 
         self.points = GeoSeries([Point(x, y) for (x, y) in zip(xs, ys)])
-        self.df = GeoDataFrame({"geometry": self.points, "x": xs, "y": ys,
-                                "s": np.zeros(num_points, dtype="object")})
+        self.df = GeoDataFrame(
+            {
+                "geometry": self.points,
+                "x": xs,
+                "y": ys,
+                "s": np.zeros(num_points, dtype="object"),
+            }
+        )
 
         self.tmpdir = tempfile.mkdtemp()
-        self.series_filename = os.path.join(self.tmpdir, "series" + ext)
-        self.frame_filename = os.path.join(self.tmpdir, "frame" + ext)
-        self.points.to_file(self.series_filename, driver=driver)
-        self.df.to_file(self.frame_filename, driver=driver)
+        self.filename = os.path.join(self.tmpdir, "frame" + self.ext)
+        self.writer(self.df, self.filename)
 
-    def teardown(self, ext):
+    def teardown(self, file_format):
         shutil.rmtree(self.tmpdir)
 
-    def time_write_frame(self, ext):
-        driver = self.driver_dict[ext]
-        with tempfile.TemporaryDirectory() as tmpdir:
-            out_filename = os.path.join(tmpdir, "frame" + ext)
-            self.df.to_file(out_filename, driver=driver)
 
-    def time_write_series(self, ext):
-        driver = self.driver_dict[ext]
+class BenchFrame(Bench):
+
+    params = ["ESRI Shapefile", "GeoJSON", "GPKG", "Parquet", "Feather"]
+    param_names = ["file_format"]
+
+    def time_write(self, file_format):
         with tempfile.TemporaryDirectory() as tmpdir:
-            out_filename = os.path.join(tmpdir, "series" + ext)
-            self.points.to_file(out_filename, driver=driver)
+            out_filename = os.path.join(tmpdir, "frame" + self.ext)
+            self.writer(self.df, out_filename)
+
+    def time_read(self, file_format):
+        self.reader(self.filename)
+
 
-    def time_read_frame(self, ext):
-        frame = GeoDataFrame.from_file(self.frame_filename)
+class BenchSeries(Bench):
```

Comment:
I kept this as a separate class for now (those don't work with Feather/Parquet), but honestly I am not sure it is very useful to keep them .. In the end it is using the same read/write functionality, the only difference is selecting only the geometry column before/after writing/reading.